### DISCUSSION TITLE
MIDI File Creation Example

### DISCUSSION THREAD

**Discussion by 9SL9** [orchetect/swift-midi#206] [Ideas]
Hi,

I'm currently use this to create midi files from scratch in my app: https://github.com/matsune/MidiParser

I'm wanting to migrate to your library (as I'm already using it elsewhere) to reduce dependencies. I'm hoping you can add some more example usage much like this project has. Specifically, I'm keen to know how to add note(s) with specific start/end times, marker(s) and tracks and to save this as a new midi file.

Thanks in advance.

**Comment by orchetect** [comment]
MIDI file documentation is definitely on the to-do list, along with an overhaul of the MIDI file code. There are no immediate plans or timelines for this however.
